spec/007_findable_songs_spec.rb

```diff
@@ -1,21 +1,35 @@
-describe 'Findable Songs' do
-  describe ".find_by_name" do
-    it 'finds a song instance in @@all by the name property of the song' do
-      song = Song.new("In the Aeroplane Over the Sea")
-      song.save
+context "Song" do
+  let!(:song) { Song.create("In the Aeroplane Over the Sea") }
 
-      expect(Song.find_by_name("In the Aeroplane Over the Sea")).to eq(song)
+  describe ".find_by_name" do
+    it "finds a song instance in @@all by the name property of the song" do
+      expect(Song.find_by_name("In the Aeroplane Over the Sea")).to be(song)
     end
   end
 
   describe ".find_or_create_by_name" do
-    it 'finds or creates a song by name maintaining uniqueness of objects by their name property' do
-      song_1 = Song.find_or_create_by_name("In the Aeroplane Over the Sea")
-      song_2 = Song.find_or_create_by_name("In the Aeroplane Over the Sea")
+    it "finds (does not recreate) an existing song with the provided name if one exists in @@all" do
+      same_song = Song.find_or_create_by_name("In the Aeroplane Over the Sea")
+
+      expect(same_song).to be(song)
```

Comment:
So this is really just testing the return of the `find_or_create_by_name` function. Maybe include a double check the song.all method for only one thing.